develop-docs/sdk/telemetry/logs.mdx

```diff
@@ -0,0 +1,381 @@
+
+API wise the SDKs are required to expose logging methods which are to be defined in a `logger` module or namespace. The SDKs should also include some initialization options to configure the behavior of logs in the SDK.
+
+### Initialization Options
+
+The SDKs need to expose the following two configuration options:
+
+- `enableSentryLogs`/`enable_sentry_logs`: A boolean flag to control if log envelopes will be generated and sent to Sentry via the Sentry SDK's Logging APIs. If this flag is set to `false`, the SDK should not send logs to Sentry. Defaults to `false`.
+
+- `beforeSendLog`/`before_send_log`: A function that takes a log object and returns a log object. This function is called before sending the log to Sentry. It can be used to modify the log object or to prevent the log from being sent to Sentry.
+
+```js
+Sentry.init({
+  enableSentryLogs: true,
+
+  beforeSendLog(log) {
+    // Prevent logs from being sent to Sentry if the plan type is enterprise
+    if (log.attributes["plan.type"] === "enterprise") {
+      return null;
+    }
+
+    return log;
+  },
+});
+```
+
+At the current moment, logs are not sampled so no sampling configuration needs to be exposed. This may change in the future.
+
+### Logger Module
+
+At minimum the SDK needs to implement the following two items to be considered to have a complete logging public API:
+
+1. `Sentry.logger.X` (where `X` is the log level): The log levels are `trace`, `debug`, `info`, `warn`, `error`, and `fatal`, which is specced out by the [protocol below](#log-severity-level).
+
+- `Sentry.logger.trace`
+- `Sentry.logger.debug`
+- `Sentry.logger.info`
+- `Sentry.logger.warn`
+- `Sentry.logger.error`
+- `Sentry.logger.fatal`
+
+These methods accepts a string template and the parameters to that string template so the SDK can perform structured logging. Optionally these methods take arbitrary attributes, but not all languages can support both passing a parameterized template and attributes in an easy way.
+
+```js
+// Need to use `fmt` helper function in JavaScript for structured logging.
+Sentry.logger.info(Sentry.logger.fmt`Adding item ${itemId} for user ${userId}`);
+
+Sentry.logger.warn("User %s performed action %s", [userId, actionName], {
+  extra: "123",
+});
+```
+
+Optionally the SDK can define a `log` log severity level, which maps to the `info` log severity level in Sentry. If an SDK collects `log` severity level, it should map to `info` severity level and `9` severity number.
+
+```js
+Sentry.logger.log("Added item to shopping cart");
+```
+
+2. `Sentry.logger.emit`: A more verbose version of the `Sentry.logger.X` methods that accepts a log object/dictionary for maximum customizability. This is only meant to be used by power users, who want full control over what exactly is being set. Note that this API aligns with [OTEL's default logging API](https://opentelemetry.io/docs/specs/otel/logs/api/#emit-a-logrecord) as well, which gives us flexibility to align on those in the future.
```

Comment:
We should not have two ways of emitting logs (one way is using the `trace`, `debug`, `info`, ... functions mentioned above) and one would be the `emit` function.

Lets not have `emit` from the start. Its addressing a problem we do not yet know people will have. We can always add it later.

Reply:
I agree now that I think about it, less surface area is generally better, one way of doing things vs. two.

> accepts a log object/dictionary for maximum customizability

We accept a log object with `trace`, `debug`, etc. anyway so maybe we just forego `emit` and leave it as private api.

Reply:
Okay I agree - removing `emit`.

Done with 9d7302bd7325cc81a63266f03ad0b1affd9b43f0